testplan/testing/base.py

```diff
@@ -441,34 +445,110 @@ def add_pre_resource_steps(self) -> None:
         self._add_step(self._build_environment)
         self._add_step(self._set_dependencies)
 
+    def _dry_run_start_resource(self) -> TestCaseReport:
+        suite_report = self.result.report[ResourceHooks.environment_start]
+        case_report = TestCaseReport(
+            name=ResourceHooks.starting,
+            category=ReportCategories.SYNTHESIZED,
+        )
+        suite_report.append(case_report)
+        return case_report
+
+    def _start_resource(self) -> TestCaseReport:
+        case_report = self._dry_run_start_resource()
+        case_result = self.cfg.result(
+            stdout_style=self.stdout_style, _scratch=self.scratch
+        )
+        self.resources.start()
+
+        for uid, driver in self.resources.items():
+            case_result.log(
+                f"{driver.__class__.__name__}[{uid}] Status: {driver.status.tag}"
+            )
+        case_report.extend(case_result.serialized_entries)
+        case_report.attachments.extend(case_result.attachments)
+        case_report.pass_if_empty()
+        return case_report
+
+    def _dry_run_stop_resource(self) -> TestCaseReport:
+        suite_report = self.result.report[ResourceHooks.environment_stop]
+        case_report = TestCaseReport(
+            name=ResourceHooks.stopping,
+            category=ReportCategories.SYNTHESIZED,
+        )
+        suite_report.append(case_report)
+        return case_report
+
+    def _stop_resource(self, is_reversed=False) -> TestCaseReport:
+        case_report = self._dry_run_stop_resource()
+        case_result = self.cfg.result(
+            stdout_style=self.stdout_style, _scratch=self.scratch
+        )
+        self.resources.stop(is_reversed)
+
+        for uid, driver in self.resources.items():
+            case_result.log(
+                f"{driver.__class__.__name__}[{uid}] Status: {driver.status.tag}"
+            )
+
+        case_report.extend(case_result.serialized_entries)
+        case_report.attachments.extend(case_result.attachments)
+        case_report.pass_if_empty()
+        return case_report
+
+    def _add_resource_report(self, name):
```

Comment:
if no resources && no start/stop related hook, skip synthesize this suite report?